### DISCUSSION TITLE
RNode & Sideband Bluetooth Setup

### DISCUSSION THREAD

**Discussion by markqvist** [markqvist/Reticulum#173] [Show and tell]
### Using RNodes in Sideband over Bluetooth

I realise the pairing / connection setup might be a little opaque currently, so let me try to outline the correct steps, and some relevant caveats here.

- The 1.x versions of the firmware only allows *one* host connection at a time. You cannot be connected to both USB/Serial from a computer, and Bluetooth from another device at the same time. The RNode will switch the host connection to only use **one** at a time. If using the device over Bluetooth, just give it power, either via the USB port or a battery. You can plug it into a computer USB port just fine, but don't open it with `rnsd` or anything else ;)

### Pairing procedure
To use an RNode over Bluetooth with an Android device, a normal Bluetooth pairing must be performed between the devices first. Here is the correct procedure:

- Connect the RNode to your Android device over USB. Wait for the Android permission dialog to appear, and grant Sideband the USB permission for the port. If this permission is not granted *to Sideband specifically*, the app will not be able to talk to the port.
- In the Sideband app, go to **Hardware** > **RNode** and press the **Enable Bluetooth** button. The Bluetooth symbol on the RNode display should now become active.
- Then press the **Start Pairing Mode** button. The Bluetooth symbol should change to indicate that the RNode is now in pairing mode.
- Go to the Bluetooth settings of your Android device and start pairing a new device. The RNode should be discovered, and when you select it for pairing, the RNode will display it's pairing code on the display. Verify that the code matches and accept the pairing from your Android device.
- The pairing procedure is now done, and Sideband will be able to use the RNode.
- You do not need to manually connect the RNode in the Bluetooth settings of your device when you need to use it, in fact this will do nothing. Sideband will initiate the connection itself through the operating system when it wants to use the RNode.

### Alternate Pairing Procedure

You can also initiate the pairing mode of the RNode from a computer using the `rnodeconf` program. Simply connect your RNode to the computer and run:

`rnodeconf --bluetooth-pair /dev/ttyUSB0`

Replace the port with the one your RNode is actually connected to, of course.

### Setup in Sideband
To use the paired RNode in Sideband, the following steps must be carried out:

- Go to **Hardware** > **RNode** and make sure that valid radio parameters have been configured. The app provides sensible defaults for most parameters, but you will at least have to specify a valid frequency for the device.
- Make sure that the **Connect via Bluetooth** checkbox is turned on.
- Go to the **Connectivity** part of the program and make sure that the **Connect via RNode** checkbox is turned on. You can optionally specify IFAC parameters for your network here.
- Shut down Sideband completely (Menu > Shutdown) and restart it. You should now see Sideband starting and connecting to your RNode.
- In the **Connectivity Status** dialog, you should be able to see the RNode listed as *on-air*.

That's it! Hope this helps anyone who've had trouble with it.

**Comment by miguelito9** [comment]
I would like to arrange an interview by Ernest Hancock with you to discuss the future of Reticulum, probably starting with some history. Ernie recently again interviewed Jason Crowe of Qortal who often mentions Reticulum as the possible non-ISP future of Qortal 
https://www.freedomsphoenix.com/Media/336392-2022-11-29-2022-11-29-ernest-interviews-jason-crowe-qortal-mp3-loaded.htm
Ernie and his partner Donna communicate by email, and currently telegram. If you would be interested in an interview, please send your email or telegram contact to Donna Hancock at producer@declareyourindependencewithernesthancock.com

Your work on Reticulum is very much appreciated and I will begin using it soon, in preparation for use with Qortal when ready.

Wishing you an amazing new year

**Reply by miguelito9** [reply]
Mark, After learning you also run the RNode project also, Ernie and Donna Hancock have both emailed you with an interview invitation and several days and times to choose from. The email subject is RNode Interview. Happy New Year

**Reply by markqvist** [reply]
Yep, that would be me as well :) I'll get in touch with them shortly.
